### ISSUE TITLE
Unable to tolerate datanode failure if it failed before the region stat was reported.

### ISSUE THREAD

**Issue by WenyXu** [GreptimeTeam/greptimedb#1762]
### What type of bug is this?

Unexpected error

### What subsystems are affected?

Datanode

### What happened?

Unable to tolerate datanode failure if it failed before the region stat was reported

### What operating system did you use?

Ubuntu Server 23.04

### Relevant log output and stack trace

e2e (killed datanode.1)
```
2023-06-12T09:11:45.179413Z  INFO chaos_master::tests::simple_region_failover: Creating table
2023-06-12T09:11:45.397666Z DEBUG chaos_master::tests::simple_region_failover: table region map: {2: [1], 1: [0]},all datanodes: [Datanode(0), Datanode(1), Datanode(2)]
2023-06-12T09:11:45.400972Z  WARN chaos_master::cluster::bare::process: pid: 3130189 exited with status: signal: 9 (SIGKILL)
2023-06-12T09:11:45.401021Z  WARN chaos_master::cluster::bare::node: node: datanode.1 , pid: 3130189, exited with reason: Ok(ExitStatus(unix_wait_status(9)))    
```

datanode.1 (failed at 2023-06-12T09:11:45.401021Z)
```
[2m2023-06-12T09:11:45.223078Z[0m [32m INFO[0m [2mtable_procedure::create[0m[2m:[0m On engine create table my_table, done, sub_id: fa725d31-3103-4b11-b2ea-8d5020d87112
[2m2023-06-12T09:11:45.231043Z[0m [32m INFO[0m [2mcommon_procedure::local::runner[0m[2m:[0m Procedure table-procedure::CreateTableProcedure-e3ff8796-2ed1-4c47-a8be-a12aefea56bb done
[2m2023-06-12T09:11:45.232169Z[0m [32m INFO[0m [2mcommon_procedure::local::runner[0m[2m:[0m Runner table-procedure::CreateTableProcedure-e3ff8796-2ed1-4c47-a8be-a12aefea56bb exits

```

datanode.2
```
[2m2023-06-12T09:11:45.265078Z[0m [32m INFO[0m [2mtable_procedure::create[0m[2m:[0m On engine create table my_table, done, sub_id: beb2f9a3-deb9-482a-9815-d150f92227f8
[2m2023-06-12T09:11:45.272961Z[0m [32m INFO[0m [2mcommon_procedure::local::runner[0m[2m:[0m Procedure table-procedure::CreateTableProcedure-53ab3d16-5830-42b1-bed5-69083fabd307 done
[2m2023-06-12T09:11:45.275175Z[0m [32m INFO[0m [2mcommon_procedure::local::runner[0m[2m:[0m Runner table-procedure::CreateTableProcedure-53ab3d16-5830-42b1-bed5-69083fabd307 exits
[2m2023-06-12T09:11:48.337100Z[0m [32m INFO[0m [2mdatanode::heartbeat[0m[2m:[0m Sending Heartbeat Request: HeartbeatRequest { header: None, peer: Some(Peer { id: 2, addr: "0.0.0.0:14102" }), is_leader: false, report_interval: None, node_stat: Some(NodeStat { rcus: 0, wcus: 0, table_num: 0, region_num: 2, cpu_usage: 0.0, load: 0.0, read_io_rate: 0.0, write_io_rate: 0.0, attrs: {} }), region_stats: [RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }, RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }], replica_stats: [], mailbox_message: None }
[2m2023-06-12T09:11:53.398511Z[0m [32m INFO[0m [2mdatanode::heartbeat[0m[2m:[0m Sending Heartbeat Request: HeartbeatRequest { header: None, peer: Some(Peer { id: 2, addr: "0.0.0.0:14102" }), is_leader: false, report_interval: None, node_stat: Some(NodeStat { rcus: 0, wcus: 0, table_num: 0, region_num: 2, cpu_usage: 0.0, load: 0.0, read_io_rate: 0.0, write_io_rate: 0.0, attrs: {} }), region_stats: [RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }, RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }], replica_stats: [], mailbox_message: None }
[2m2023-06-12T09:11:58.458454Z[0m [32m INFO[0m [2mdatanode::heartbeat[0m[2m:[0m Sending Heartbeat Request: HeartbeatRequest { header: None, peer: Some(Peer { id: 2, addr: "0.0.0.0:14102" }), is_leader: false, report_interval: None, node_stat: Some(NodeStat { rcus: 0, wcus: 0, table_num: 0, region_num: 2, cpu_usage: 0.0, load: 0.0, read_io_rate: 0.0, write_io_rate: 0.0, attrs: {} }), region_stats: [RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }, RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }], replica_stats: [], mailbox_message: None }
[2m2023-06-12T09:12:03.518261Z[0m [32m INFO[0m [2mdatanode::heartbeat[0m[2m:[0m Sending Heartbeat Request: HeartbeatRequest { header: None, peer: Some(Peer { id: 2, addr: "0.0.0.0:14102" }), is_leader: false, report_interval: None, node_stat: Some(NodeStat { rcus: 0, wcus: 0, table_num: 0, region_num: 2, cpu_usage: 0.0, load: 0.0, read_io_rate: 0.0, write_io_rate: 0.0, attrs: {} }), region_stats: [RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }, RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }], replica_stats: [], mailbox_message: None }
[2m2023-06-12T09:12:08.578386Z[0m [32m INFO[0m [2mdatanode::heartbeat[0m[2m:[0m Sending Heartbeat Request: HeartbeatRequest { header: None, peer: Some(Peer { id: 2, addr: "0.0.0.0:14102" }), is_leader: false, report_interval: None, node_stat: Some(NodeStat { rcus: 0, wcus: 0, table_num: 0, region_num: 2, cpu_usage: 0.0, load: 0.0, read_io_rate: 0.0, write_io_rate: 0.0, attrs: {} }), region_stats: [RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }, RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }], replica_stats: [], mailbox_message: None }
[2m2023-06-12T09:12:13.638395Z[0m [32m INFO[0m [2mdatanode::heartbeat[0m[2m:[0m Sending Heartbeat Request: HeartbeatRequest { header: None, peer: Some(Peer { id: 2, addr: "0.0.0.0:14102" }), is_leader: false, report_interval: None, node_stat: Some(NodeStat { rcus: 0, wcus: 0, table_num: 0, region_num: 2, cpu_usage: 0.0, load: 0.0, read_io_rate: 0.0, write_io_rate: 0.0, attrs: {} }), region_stats: [RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }, RegionStat { region_id: 4402341478401, table_name: Some(TableName { catalog_name: "greptime", schema_name: "public", table_name: "my_table" }), rcus: 0, wcus: 0, approximate_bytes: 0, approximate_rows: 0, attrs: {"engine_name": "mito"} }], replica_stats: [], mailbox_message: None }

```

### How can we reproduce the bug?
1. Create a multiple regions table
2. Kill one of datanode containing the table region immediately (before heartbeat reports the table stat)

For Internal users: run this [test (comment the highlight lines)](https://github.com/GreptimeTeam/greptimedb-chaos/blob/e809ca954f9a3c61316004ff1d9a57c61c5bca3a/src/chaos/src/tests/simple_region_failover.rs#L148-L150)

**Comment by MichaelScofield** [comment]
On second thought, I don't think we need to fix it now. This issue boils down to the problem that what to do if the region's running stat is unknown to metasrv. But first, is it possible? 

The region stat could be unknown to metasrv for two reasons: 

1. the datanode is not started at all. But it's cloud era, in k8s, the datanode pods in stateful set can be seen as "always on". 2. the region is corrupted, can't be opened in datanode. In this case, it's meaningless to do region failover - the region can't be opened in another datanode either. 

So, in conclusion, any health regions can be opened, and will submit stat to metasrv via datanode's heartbeat. They can be properly governed by current failure detector. WDYT @WenyXu @fengjiachun @Fengys123

**Comment by fengjiachun** [comment]
> On second thought, I don't think we need to fix it now. This issue boils down to the problem that what to do if the region's running stat is unknown to metasrv. But first, is it possible?
> 
> The region stat could be unknown to metasrv for two reasons:
> 
> 1. the datanode is not started at all. But it's cloud era, in k8s, the datanode pods in stateful set can be seen as "always on".
> 2. the region is corrupted, can't be opened in datanode. In this case, it's meaningless to do region failover - the region can't be opened in another datanode either.
> 
> So, in conclusion, any health regions can be opened, and will submit stat to metasrv via datanode's heartbeat. They can be properly governed by current failure detector. WDYT @WenyXu @fengjiachun @Fengys123

I agree with @MichaelScofield's viewpoint that metasrv must schedule datanode dispatch based on heartbeat data(region stat).

**Comment by WenyXu** [comment]
> On second thought, I don't think we need to fix it now. This issue boils down to the problem that what to do if the region's running stat is unknown to metasrv. But first, is it possible?
> 
> The region stat could be unknown to metasrv for two reasons:
> 
> 1. the datanode is not started at all. But it's cloud era, in k8s, the datanode pods in stateful set can be seen as "always on".
> 2. the region is corrupted, can't be opened in datanode. In this case, it's meaningless to do region failover - the region can't be opened in another datanode either.
> 
> So, in conclusion, any health regions can be opened, and will submit stat to metasrv via datanode's heartbeat. They can be properly governed by current failure detector. WDYT @WenyXu @fengjiachun @Fengys123

What if it happens the during the scaling down? I think we need carefully design the case of scaling down.

**Comment by MichaelScofield** [comment]
@WenyXu on datanode scaling down, there must have another active maintenance procedure to move regions between datanodes.
